app/tests/test_training.py

```diff
@@ -648,6 +651,97 @@ def test_difficulty_form(self):
 
                 # should not create any new attempts
                 self.assertEqual(Attempt.query.count(), 1)
+    
+    def test_correct_fill_in_the_blank_question_submission(self):
+        self.course.users.append(self.u2)
+        db.session.commit()
+
+        for user in [self.u1, self.u2]:
+            client = self.app.test_client(user=user)
+            response = client.post(url_for('user_views.test',
+                                                course_name="test-course",
+                                                mission_id=1),
+                                        data={
+                                            "question_id": str(self.fitb_question.id),
+                                            "response": "Answer1, Answer2",
+                                            "booger": "MEOW",
+                                            "submit": "y"
+                                        })
+
+            # check that user was sent to the difficulty page
+            self.assertEqual(response.status_code, 302)
+            self.assertEqual(urlparse(response.location).path,
+                                      url_for('user_views.difficulty',
+                                              course_name="test-course",
+                                              mission_id=1,
+                                              _external=False))
+
+            # check that there is now a single (text) attempt
+            self.check_text_attempt(self.fitb_question.id, user.id, "Answer1, Answer2")
+
+            attempt = TextAttempt.query.first()
+            self.assertTrue(attempt.correct)
+
+            # clear out attempts for next user test
+            Attempt.query.delete()
+            TextAttempt.query.delete()
+
+    def test_incorrect_fill_in_the_blank_question_submission(self):
```

Comment:
Again, I think we need to be a bit more rigorous with our unit testing.

The current "incorrect" test has a single response when there are two expected. This shouldn't even be possible because our JS code would still add a "blank" part if they didn't fill it in.

I would have cases where they try "Answer1, WrongAnswer2" and "WrongAnswer1, Answer2" to make sure that they both are reject as incorrect answers.